src/app/modules/submission/submission.middleware.ts

```diff
@@ -1,14 +1,66 @@
 import { RequestHandler } from 'express'
 import { ParamsDictionary } from 'express-serve-static-core'
+import { StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
 
 import { createLoggerWithLabel } from '../../../config/logger'
 import { WithAutoReplyEmailData } from '../../../types'
+import getSubmissionModel from '../../models/submission.server.model'
 
 import * as SubmissionService from './submission.service'
 import { ProcessedFieldResponse } from './submission.types'
 
+const SubmissionModel = getSubmissionModel(mongoose)
+
 const logger = createLoggerWithLabel(module)
 
+/**
+ * Verifies that by saving this submission, the submission count does not exceed the maximum
+ * specified submissions available (if any).
+ */
+export const checkSubmissionCountBelowMaximum: RequestHandler<
+  { formId: string },
+  unknown,
+  EncryptSubmissionBody
+> = (req, res, next) => {
+  const { form } = req as WithForm<typeof req>
+
+  if (form.submissionLimit !== -1) {
+    return SubmissionModel.countDocuments({ form: form._id })
+      .exec()
+      .then((queryResults) => {
+        if (queryResults < form.submissionLimit) {
+          return next()
+        } else {
+          logger.error({
+            message:
+              'Current submission exceeded maximum allowable submissions.',
+            meta: { form: form._id },
+          })
+          return res.status(StatusCodes.FORBIDDEN).json({
```

Comment:
Suggest using HTTP 429 Too Many Requests instead - HTTP 403 Forbidden implies the requests were rejected because of authorisation problems.